pkg/collector/corechecks/cluster/ksm/kubernetes_state.go

```diff
@@ -326,6 +330,35 @@ func (k *KSMCheck) Configure(integrationConfigDigest uint64, config, initConfig
 	return nil
 }
 
+func filterUnknownCollectors(client discovery.DiscoveryInterface, collectors []string) ([]*v1.APIResourceList, []string, error) {
+	resourcesSet := make(map[string]struct{}, len(collectors))
+	resources, err := client.ServerResources()
+	if err != nil {
+		if !discovery.IsGroupDiscoveryFailedError(err) {
+			return nil, nil, fmt.Errorf("unable to perform resource discovery: %s", err)
+		} else {
+			for group, apiGroupErr := range err.(*discovery.ErrGroupDiscoveryFailed).Groups {
+				log.Warnf("unable to perform resource discovery for group %s: %s", group, apiGroupErr)
+			}
+		}
+	}
+	for _, resourceList := range resources {
+		for _, resource := range resourceList.APIResources {
+			resourcesSet[resource.Name] = struct{}{}
+		}
+	}
```

Comment:
I would suggest to extract from this method the `resources` retrival. and pass it as argument of the `filterUnknownCollectors` function

Reply:
Indeed, it would be better. It is done now and tested.